doc/UsersGuide/source/ContributingDevelopment.rst

```diff
@@ -13,4 +13,284 @@ Contributing Development
+
+It is suggested that the developers inform all the related code managers as the hierarchy structure of the ufs-weather-model repository may require collaboration among the code managers.
+
+-----------------------------------
+Engaging in the code review process
+-----------------------------------
+
+When code managers receive a pull request to commit the code changes, it is recommended that they add at least two code reviewers to review the code. The reviewers will write comments about the code changes and give a recommendation as to whether the code changes can be committed. What kinds of code changes will be accepted in the repository is beyond the scope of this document; future ufs-weather-model code management may have detailed answer for that.
+
+Reviewers may suggest some code changes during the review process. Developers need to respond to these comments in order to get code changes committed. If developers make further changes to their branch, reviewers need to check the code changes again. When both reviewers give recommendation to commit the code, code managers will merge the changes into the repository.
+
+---------------------------
+Conducting regression tests
+---------------------------
+
+Only developers that are running on a limited set of platforms (Hera, Cheyenne, WCOSS) can compile and run regression tests using the ufs-weather-model.
+
+To run regression test using rt.sh
+
+.. code-block:: console
+
+   % cd ufs-weather-model/tests
+   % ./rt.sh -f
+
+Regression test log files (ufs-weather-model/tests/Compile_$(MACHINE_ID).log and ufs-weather-model/tests/RegressionTests_$(MACHINE_ID).log ) will be updated.
+
+To create new baseline:
+
+.. code-block:: console
+
+   % cd ufs-weather-model/tests
+   % ./rt.sh -f -c
+
+      * To run regression test using NEMSCompsetRun
+
+.. code-block:: console
+
+   % cd ufs-weather-model
+   % ./NEMS/NEMSCompsetRun -f 
+
+Regression test log files (ufs-weather-model/log/$MACHINE_ID/* ) will be updated.
+
+To create new baseline:
+
+.. code-block:: console
+
+   % cd ufs-weather-model
+   % ./NEMS/NEMSCompsetRun--baseline fv3 --platform=${PLATFORM}
+
+
+The value of ${PLATFORM} can be found in ufs-weather-model/compsets/platforms.input.
+
+Developers need to commit the regression test log files to their branch before making pull request.
+
+-------------------------------------
+Compiling the code and running a test
+-------------------------------------
+
+Currently developers running on Hera, Cheyenne or WCOSS can compile and run tests using the ufs-weather-model. In the document below, ufs-weather-model directory points to a branch in the developer’s personal fork.
```

Comment:
Instead of listing specific platforms, just say Tier 1 and Tier 2 machines and point reader to platform policy in wiki. That way this documentation will not age off as more machines are added.
https://github.com/ufs-community/ufs-weather-model/wiki/Weather-Model-Platform-and-Compiler-Support